### ISSUE TITLE
Suggestion: SQLiteAppender implement setMaxHistory

### ISSUE THREAD

**Issue by mauron85** [tony19/logback-android#123]
Hi, firstly I want to say this is really great library. Already using [similar library](https://github.com/CocoaLumberjack/CocoaLumberjack/) for iOS in my [open source project](https://github.com/mauron85/cordova-plugin-background-geolocation/tree/next-android) and I'm about to use `logback-android` for android logging. I'm specially interested into logging to SQLite db.

The only thing I'm missing is `setMaxHistory` in days similar to `RollingFileAppender`, so debug logs won't get too big. Will you add this? Many thanks.

**Comment by tony19** [comment]
Sure, I can look into it this weekend.

**Comment by mauron85** [comment]
@tony19 Any updates?

**Comment by tony19** [comment]
@mauron85 Apologies for the delay. I will tackle this later today. (for real 😉 )

**Comment by tony19** [comment]
The `SQLiteAppender` will be modified to create a trigger that deletes the oldest entry (or entries) so that the remaining row count equals `maxHistory`. For example, if `maxHistory` were set to 1000, this trigger is added to the SQLite database created at `/data/data/<packagename>/databases/logback.db`:

```
CREATE TRIGGER logback_roll AFTER INSERT ON logging_event
BEGIN
  DELETE FROM logging_event
  WHERE ROWID <= (SELECT ROWID FROM logging_event ORDER BY ROWID DESC LIMIT 1000, 1);
END;
```

The trigger is recreated on each startup of the appender.

You could manually add this trigger to your `logback.db` (e.g., running `sqlite3` via adb shell), and it should perform the rollover on the next logging event.

**Comment by mauron85** [comment]
Seems nice and effective to me. Not sqlite expert but maybe instead rowid some kind of timestamp can be used in delete statement.

**Comment by tony19** [comment]
That could be done as there is a timestamp column, but row ID is simpler IMO.

**Comment by mauron85** [comment]
Also is it possible to have maxHistory expressed in days not in number of log entries?

**Comment by tony19** [comment]
Ok, I've scrapped the trigger design (which I think would be inefficient). `<maxHistory>` is now a time duration string (e.g., `12 hours`, `2 days`, etc.) that gets applied upon a logging event only if the last clearing occurrence exceeded the given duration.

For example, given `<maxHistory>1 day</maxHistory>`, and the following logging events:

- 1 Oct 2016 @ 12:00pm - *appender started...clears expired logs*
- 1 Oct 2016 @ 3:30pm - *last action < 1 day...nothing to do*
- 2 Oct 2016 @ 10:00am - *last action < 1 day...nothing to do*
- 2 Oct 2016 @ 5:15pm - *last action >= 1 day...clears expired logs*

This behavior is similar to how `RollingFileAppender` clears expired files.

**Comment by tony19** [comment]
Released in [1.1.1-6](https://github.com/tony19/logback-android/releases/tag/v_1.1.1-6)
